frontend/packages/kubevirt-plugin/src/components/boot-order/boot-order.tsx

```diff
@@ -0,0 +1,144 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { DNDList } from '../dnd-list';
+import { Empty } from './empty';
+import { AddItemRowFactory } from './add-item-row-factory';
+
+const addItemMessage = 'Add source';
+const addItemDisabledMessage = 'All sources selected';
+
+// Boot order will display a DNDList populated by devices.sources,
+// with a list of optional devices for adding.
+//
+// it has 3 states:
+// 1 - no sources selected - will show an emtpy with a call for action.
+// 2 - regular mode        - show a list of sources, with a button to enter edit mode.
+// 3 - edit mode           - show a list of sources, with a select line for new source.
+//
+// Props <BootOrderProps>:
+//   devices <BootOrderDrvices>:
+//       sources any[]: ordered list of devices with bootOrder set.
+//       options any[]: a list of optional devices to set as bootable.
+//   setDevice: a react state method to set the devices.
+//
+// Usage:
+//
+//   import { getBootableDevicesInOrder, getUnBootableDevices } from '.../selectors/vm/combined';
+//   ...
+//   const [devices, setDevices] = React.useState<BootOrderDrvices>({
+//     sources: _.cloneDeep(getBootableDevicesInOrder(vmLikeEntity)),
+//     options: _.cloneDeep(getUnBootableDevices(vmLikeEntity)),
+//   });
+//   ...
+//   <BootOrder devices={devices} setDevices={setDevices} />
+export const BootOrder = ({ devices, setDevices }: BootOrderProps) => {
+  const [state, setState] = React.useState<BootOrderState>({
+    isEditMode: false,
+  });
+
+  const setEditMode = (mode: boolean): void => {
+    setState({
+      isEditMode: mode,
+    });
+  };
+
+  // Move an item form the options list to the sources list, item key is "<type>->name>".
+  const addByKey = (key: string): void => {
+    const item = _.find(devices.options, (option) => `${option.type}-${option.value.name}` === key);
+
+    const sources = [...devices.sources, item];
+    const options = _.filter(
+      devices.options,
+      (option) => `${option.type}-${option.value.name}` !== key,
+    );
+
+    setDevices({
+      sources,
+      options: _.orderBy(options, 'type', 'value.name'),
+    });
+    setState({ isEditMode: false });
+  };
+
+  // Move item from the sources list to the options by index.
+  const onDelete = (index: number) => {
+    const item = devices.sources[index];
+
+    const sources = [...devices.sources.slice(0, index), ...devices.sources.slice(index + 1)];
+    const options = [...devices.options, item];
+
+    setDevices({
+      sources,
+      options: _.orderBy(options, ['type', 'value.name']),
```

Comment:
orderBy could be done only just when rendering

Reply:
done !